### PR TITLE
INA266 Power Monitor Firmware

### LINKED ISSUE #1992: Power monitor firmware
### Description of the task
Write abstractions around I2C communication to the INA266 to get information from the power monitor

### Acceptance criteria

- [ ] communication with INA266 works
- [ ] good abstractions

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
Power monitor firmware for INA266 over I2C

### Testing Done
Prints out power monitor

### Resolved Issues
resolves #1992

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by akhilveeraghanta** [comment]
lgtm! just a couple un-resolved TODOs and merging the `frankie_v1` -> `robot_stm32h7` merge and this should be good to go

File: docs/getting-started.md [hunk 1]
```diff
@@ -236,12 +236,13 @@ This will output the file at the _absolute_ path given via the `--callgrind-out-
...
 4. At this point you should be in a gdb window. Take a look at [this tutorial](https://www.cprogramming.com/gdb.html) for some basics.
+5. To flash the firmware without debugging, run `bazel run --cpu=stm32h7 --compilation_mode=dbg //firmware/tools:stm32h7_firmware 0 program`
 
 ## Working with CubeMX to regenerate code
 1. Make sure you've followed [Installing Firmware Dependencies](#installing-firmware-dependencies)
-2. To regenerate code from the `.ioc` file, run `bazel run //firmware_new/tools:cubemx_regen path/to/directory/with/.ioc`. The directory that is passed in as an argument must only contain 1 ioc file, which will be used to generate code into the same directory.
+2. To regenerate code from the `.ioc` file, run `bazel run //firmware/tools:cubemx_regen path/to/directory/with/.ioc`. The directory that is passed in as an argument must only contain 1 ioc file, which will be used to generate code into the same directory.
```
Comment (kelvinkoon) [inline docs/getting-started.md:246]: Is it worth checking if there's more than 1 `.ioc` file to abort? (Don't know about consequences if there is more than 1 file in the directory)

File: src/firmware/boards/robot_stm32h7/io/power_monitor.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ */
+void io_power_monitor_init(I2C_TypeDef *i2c_instance, uint16_t dev_address,
+                           uint16_t INA226_config);
+
+/**
+ * Gets the battery voltage
+ *
+ * @param the power monitor to get voltage with
+ *
+ * @param the current voltage of the battery in volts
```
Comment (kelvinkoon) [inline src/firmware/boards/robot_stm32h7/io/power_monitor.h]: Javadoc doesn't match function signature.

File: src/firmware/boards/robot_stm32h7/io/power_monitor.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ */
+float io_power_monitor_getBatteryVoltage(void);
+
+/**
+ * Destroy the monitor
+ *
+ * @param the power monitor to destroy
```
Comment (kelvinkoon) [inline src/firmware/boards/robot_stm32h7/io/power_monitor.h]: Likewise with function signature.

**Review by kelvinkoon — CHANGES_REQUESTED** [review]
Nice work!

File: src/firmware/boards/robot_stm32h7/io/power_monitor.h (hunk 2 above)
Comment (jonathanlew) [reply src/firmware/boards/robot_stm32h7/io/power_monitor.h]: fixed

File: docs/getting-started.md (hunk 1 above)
Comment (jonathanlew) [reply docs/getting-started.md:246]: yes the script does check that

File: src/firmware/boards/robot_stm32h7/io/power_monitor.h (hunk 3 above)
Comment (jonathanlew) [reply src/firmware/boards/robot_stm32h7/io/power_monitor.h]: fixed

Comment (jonathanlew) [reply src/firmware/boards/robot_stm32h7/io/power_monitor.h]: fixed

**Review by kelvinkoon — APPROVED** [review]
🔋

**Review by akhilveeraghanta — APPROVED** [review]
:shipit:

**Pull request merged**